string action
string target
float64 offsetX
float64 offsetY
float64 angle
int64 timestamp